mocks.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 The Prometheus Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tsdb
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/prometheus/tsdb/labels"
+	"github.com/prometheus/tsdb/testutil"
+	"github.com/prometheus/tsdb/tsdbutil"
+)
+
+// This file holds types and functions that are used for testing
+// purposes.
+
+const (
+	MockDefaultLabelName  = "labelName"
+	MockDefaultLabelValue = "labelValue"
+)
+
+type mockSeries struct {
+	labels   func() labels.Labels
+	iterator func() SeriesIterator
+}
+
+// MockCreateBlock creates a block with given set of series and returns its dir.
+// Intended for testing purposes.
+func CreateBlock(tb testing.TB, dir string, series []Series) string {
```

Comment:
No way to put this in tsdbutil package ?

Reply:
Not without massive refactoring I think. A lot of the functionality in `mocks.go` depends on stuff from the `tsdb` package: for example `CreateBlock` calls `createHead` which calls `Head()` from `head.go`.  Putting all these into `tsdbutil` would mean we have to import `tsdb` into `tsdbutil`, which is a circular import (since `tsdb` imports `tsdbutil` in a lot of places). 

To avoid this I can see three options:

1. Come up with some clever interfaces that we can implement in bock `package tsdb` (+ tests), as well as in the tests of `package main`.
2. Move all existing tests into a new package, then import from both `tsdb` and `tsdbutil`.
3. Export `CreateBlock` and `GenSeries` from `tsdb`.

I think option 1 and 2 require bigger changes in regards to architecture and refactoring and would increase the size of this PR by around x5 which might be out of scope for this PR. So I thought just leaving those functions in `tsdb` and exporting them is the least-intrusive way forward for this. What do you think?

Reply:
I am not sure, 
would be interested to see what @codesome @gouthamve 	@bwplotka 	think.